File: app/src/main/java/org/apache/fineract/data/remote/ConnectivityInterceptor.java
package org.apache.fineract.data.remote;

import android.content.Context;
import androidx.annotation.NonNull;

import org.apache.fineract.exceptions.NoConnectivityException;
import org.apache.fineract.utils.NetworkUtil;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

/**
 * @author Rajan Maurya
 *         On 23/09/17.
 */
public class ConnectivityInterceptor implements Interceptor {

    private Context context;

    public ConnectivityInterceptor(Context context) {
        this.context = context;
    }

    @Override
    public Response intercept(@NonNull Chain chain) throws IOException {
        if (!NetworkUtil.isConnected(context)) {
            throw new NoConnectivityException();
        }

        Request.Builder builder = chain.request().newBuilder();
        return chain.proceed(builder.build());
    }
}

File: app/src/main/java/org/apache/fineract/data/remote/ConnectivityInterceptor.kt
package org.apache.fineract.data.remote

import android.content.Context
import okhttp3.Interceptor
import okhttp3.Response
import org.apache.fineract.exceptions.NoConnectivityException
import org.apache.fineract.utils.NetworkUtil
import java.io.IOException

/**
 * Created by Ahmad Jawid Muhammadi on 1/6/20
 */

class ConnectivityInterceptor(private var context: Context?) : Interceptor {

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response? {
        if (!NetworkUtil.isConnected(context)) {
            throw NoConnectivityException()
        }
        val builder = chain.request().newBuilder()
        return chain.proceed(builder.build())
    }
}